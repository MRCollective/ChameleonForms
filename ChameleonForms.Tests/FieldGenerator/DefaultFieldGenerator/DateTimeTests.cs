using ApprovalTests.Html;
using ChameleonForms.Component.Config;
using ChameleonForms.Templates;
using NUnit.Framework;

namespace ChameleonForms.Tests.FieldGenerator.DefaultFieldGenerator
{
    class DateTimeTests : DefaultFieldGeneratorShould
    {
        [Test]
        public void Use_correct_html_for_datetime_field()
        {
            var g = Arrange(m => m.DateTime);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_nullable_datetime_field()
        {
            var g = Arrange(m => m.NullableDateTime);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_datetime_field_with_format()
        {
            var g = Arrange(m => m.DateTimeWithFormat);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_nullable_datetime_field_with_format()
        {
            var g = Arrange(m => m.NullableDateTimeWithFormat);

            var result = g.GetFieldHtml(new FieldConfiguration { Attributes = new HtmlAttributes(data_attr => "value") });

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }
    }
}
