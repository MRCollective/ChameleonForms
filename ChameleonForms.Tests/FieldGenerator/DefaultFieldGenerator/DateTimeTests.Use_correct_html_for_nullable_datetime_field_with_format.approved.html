<input class="input-validation-error" data-attr="value" data-val="true" data-val-date="The field NullableDateTimeWithFormat must be a date." data-val-format="d/M/yyyy" id="NullableDateTimeWithFormat" name="NullableDateTimeWithFormat" type="text" value="" />